File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) 
    {
        unordered_map<string, vector<string>> mpp;

        for(auto it : strs)
        {
            string temp = it;
            sort(temp.begin(), temp.end());
            mpp[temp].push_back(it);
        }    

        vector<vector<string>> res;
        for(auto it : mpp)
        {
            res.push_back(it.second);
        }

        return res;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:    
        res = defaultdict(list)

        for word in strs:
            mpp = [0] * 26

            for ch in word:
                mpp[ord(ch) - ord('a')] += 1
            
            res[tuple(mpp)].append(word)
        
        return list(res.values())